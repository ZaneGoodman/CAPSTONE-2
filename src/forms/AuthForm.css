.AuthForm-input {
  text-align: center;
  margin-top: 5px;
}
.AuthForm-form {
  margin: auto;
  margin-top: 50px;
  text-align: center;
  align-items: center;
  width: 20rem;
}
.AuthForm-errors {
  text-align: left;
  color: rgb(240, 149, 149);
}
.AuthForm-label {
  font-family: "Rosarivo", cursive;
  color: rgb(236, 234, 226);
}
.AuthForm-btn {
  font-family: "Rosarivo", cursive;
  margin: 5px;
}
