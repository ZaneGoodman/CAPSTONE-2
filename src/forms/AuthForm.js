import React, { useState } from "react";
import { useFormik } from "formik";
import * as Yup from "yup";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import "./AuthForm.css";
/**Login user */

const AuthForm = (props) => {
  const [valid, setValid] = useState(true);
  const checkLogin = async (values) => {
    //Call login function, if it returns invalid show error message
    if ((await props.login(values)) === "invalid") {
      setValid(false);
    } else {
      await props.login(values);
    }
  };
  const formik = useFormik({
    initialValues: {
      username: "",
      password: "",
    },
    validationSchema: Yup.object({
      username: Yup.string()
        .max(25, "Must be 25 Characters or less")
        .required("Required"),
      password: Yup.string().required("Required"),
    }),
    onSubmit: (values) => {
      checkLogin(values);
    },
  });

  return (
    <>
      <Form className="AuthForm-form">
        <Form.Group className="mb-3">
          <Form.Label htmlFor="username" className="AuthForm-label">
            Username
          </Form.Label>
          <Form.Control
            className="AuthForm-input"
            onChange={formik.handleChange}
            type="text"
            autoComplete="username"
            placeholder="......."
            id="username"
            name="username"
            onBlur={formik.handleBlur}
            value={formik.values.username}
          />
          {formik.touched.username && formik.errors.username ? (
            <p className="AuthForm-errors">{formik.errors.username}</p>
          ) : null}
        </Form.Group>
        <Form.Group className="mb-3">
          <Form.Label className="AuthForm-label">Password</Form.Label>
          <Form.Control
            className="AuthForm-input"
            onChange={formik.handleChange}
            type="password"
            autoComplete="current-password"
            placeholder="......."
            id="password"
            name="password"
            onBlur={formik.handleBlur}
            value={formik.values.password}
          />
          {formik.touched.password && formik.errors.password ? (
            <p className="AuthForm-errors">{formik.errors.password}</p>
          ) : null}
          {valid === false ? (
            <p className="AuthForm-errors">Invalid username/password</p>
          ) : null}
        </Form.Group>
        <Button
          variant="dark"
          className="AuthForm-btn"
          onClick={formik.handleSubmit}>
          Join
        </Button>
      </Form>
    </>
  );
};

export default AuthForm;
