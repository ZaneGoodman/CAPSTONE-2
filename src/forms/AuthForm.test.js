import React from "react";
import { render } from "@testing-library/react";

import AuthForm from "./AuthForm";
//Smoke test
test("renders AuthForm component with authentication", async () => {
  render(<AuthForm login={() => "logged in"} />);
});

//Snapshot test  <MemoryRouter>
it("AuthForm matches snapshot", async () => {
  const { asFragment } = render(<AuthForm login={() => "logged in"} />);
  await expect(asFragment()).toMatchSnapshot();
});
