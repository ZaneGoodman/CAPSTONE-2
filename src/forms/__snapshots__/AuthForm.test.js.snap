// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="AuthForm-form"
  >
    <div
      class="mb-3"
    >
      <label
        class="AuthForm-label form-label"
        for="username"
      >
        Username
      </label>
      <input
        autocomplete="username"
        class="AuthForm-input form-control"
        id="username"
        name="username"
        placeholder="......."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="AuthForm-label form-label"
      >
        Password
      </label>
      <input
        autocomplete="current-password"
        class="AuthForm-input form-control"
        id="password"
        name="password"
        placeholder="......."
        type="password"
        value=""
      />
    </div>
    <button
      class="AuthForm-btn btn btn-dark"
      type="button"
    >
      Join
    </button>
  </form>
</DocumentFragment>
`;
